keras_nlp/models/whisper/whisper_audio_to_speech_lm_preprocessor.py

```diff
@@ -0,0 +1,213 @@
+# Copyright 2023 The KerasNLP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Whisper AudioToSpeech LM preprocessor layer."""
+
+import copy
+
+import tensorflow as tf
+from absl import logging
+
+from keras_nlp.api_export import keras_nlp_export
+from keras_nlp.models.whisper.whisper_preprocessor import WhisperPreprocessor
+from keras_nlp.models.whisper.whisper_presets import backbone_presets
+from keras_nlp.utils.keras_utils import (
+    convert_inputs_to_list_of_tensor_segments,
+)
+from keras_nlp.utils.keras_utils import pack_x_y_sample_weight
+from keras_nlp.utils.python_utils import classproperty
+
+
+@keras_nlp_export("keras_nlp.models.WhisperAudioToSpeechLMPreprocessor")
+class WhisperAudioToSpeechLMPreprocessor(WhisperPreprocessor):
+    """Whisper AudioToSpeech LM preprocessor.
+
+    This layer is used as preprocessor for seq2seq tasks using the Whisper model.
```

Comment:
maybe use this language instead https://github.com/keras-team/keras-nlp/blob/03668b87b6aae0e4d87cda041e07e2dc81736331/keras_nlp/models/gpt2/gpt2_causal_lm_preprocessor.py#L33-L43

a little more up to date